### PR TITLE
[#333] Update dependencies

### LINKED ISSUE #333: Update to new dependencies and GHC 9
# Clarification and motivation

New versions have been released of `baseDAO`'s dependencies, in particular:
- [morley-1.17.0](https://hackage.haskell.org/package/morley-1.17.0)
- [lorentz-0.14.0](https://hackage.haskell.org/package/lorentz-0.14.0)
- [morley-client-0.2.0](https://hackage.haskell.org/package/morley-client-0.2.0)
- [cleveland-0.2.0](https://hackage.haskell.org/package/cleveland-0.2.0)

moreover:
1. one for `indigo` should be released after [indigo/#58](https://gitlab.com/morley-framework/indigo/-/issues/58 )
2. `morley-ledgers` will be up-to-date after [morley-ledgers/#25](https://gitlab.com/morley-framework/morley-ledgers/-/issues/25)
3. `morley-metadata` will be up-to-date after [morley-metadata/#15](https://gitlab.com/morley-framework/morley-metadata/-/issues/15)

_Note: this issue depends on these ones_

Consequently `baseDAO` should be updated to use them and switch to GHC 9 

# Acceptance criteria

The project is updated to support and use GHC 9 as well as the new versions of dependencies mentioned above.

### PR DESCRIPTION
**Pull request by sras**
## Description

Updated versions of Morley and friends.

## Related issue(s)

Resolves #333

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: haskell/test/Test/Ligo/RegistryDAO.hs [hunk 1]
```diff
@@ -128,9 +128,9 @@ initialStorage :: Address -> RegistryStorage
 initialStorage admin = let
   fs = baseDAORegistryStorageLigo { sExtra = def }
   in fs { sAdmin = admin, sConfig = (sConfig fs)
-            { cPeriod = 11
-            , cProposalFlushLevel = 22
-            , cProposalExpiredLevel = 100
+            { cPeriod = 25
+            , cProposalFlushLevel = 50
+            , cProposalExpiredLevel = 300
```
Comment (pasqu4le) [inline haskell/test/Test/Ligo/RegistryDAO.hs:133]: :thinking: how is this change caused by the update of the dependencies?

File: docs/building.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ in particular its CameLIGO dialect.
 To generate the contract code, or the storage for one of the provided examples,
 you will need the [ligo executable](https://ligolang.org/docs/intro/installation) installed.
 
-The latest working version tested is [0.46.0](https://gitlab.com/ligolang/ligo/-/releases/0.46.0).
+The latest working version tested is [0.47.0](https://gitlab.com/ligolang/ligo/-/releases/0.47.0).
```
Comment (pasqu4le) [inline docs/building.md:14]: This commit 9bb6a19d9e2009b263d71d3e5403e9d279c9601e only has this change, but its message mentions deprecations and warnings.

I also see that the LIGO versions used by CI hasn't been updated here.

So, I suppose that this is a rebase artifact, I'd suggest to drop this commit entirely.

**Review by pasqu4le — CHANGES_REQUESTED** [review]

File: haskell/test/Test/Ligo/RegistryDAO.hs (hunk 1 above)
Comment (sras) [reply haskell/test/Test/Ligo/RegistryDAO.hs:133]: Had to change due to CI on-chain test failures.

File: docs/building.md (hunk 2 above)
Comment (sras) [reply docs/building.md]: Dropped

**Review by pasqu4le — APPROVED** [review]

**Pull request merged**